src/main.c

```diff
@@ -288,6 +291,46 @@ static uint32_t read_encoder_right(void)
 
 
 /**
+ * @brief Setup for adc: Four injected channels on scan mode for ADC1.
+ * - Initialize channel_sequence structure to map physical channels vs
+ * sw injected channels.
+ * - Power off the ADC to be sure that doesn't run during configuration.
+ * - Enable scan mode with single conversion mode triggered by software.
+ * - Configure the alignment (right) and the sample time (28.5 cycles of ADC
+ * clock)
+ * - Set injected sequence with channel_sequence structure
+ * - Power on the ADC and wait for ADC starting up (about 3 ms, See adc_power_on
+ * function from libopencm3)
+ * - Calibrate the ADC
+ *
+ * @see Reference manual (RM0008) "Analog-to-digital converter" and in
+ * particular "Scan mode" section.
+ */
+static void setup_adc(void)
+{
+	int i;
+
+	uint8_t channel_sequence[4] = {ADC_CHANNEL4, ADC_CHANNEL5, ADC_CHANNEL6,
+	ADC_CHANNEL7};
```

Comment:
What if we used something like...?

```c
uint8_t sequence_size = 4
uint8_t channel_sequence[sequence_size] = {...};

/* [...] */
adc_set_injected_sequence(ADC1, sequence_size, channel_sequence);
```

It is true that this way you need to change 2 lines if you want to change the sequence size, but at least they are together and, most important, the compiler will complain if you do it wrong. On the bright side, we do not need to use the `sizeof` later. :thinking:

Reply:
Ok, good solution.